presto-spi/src/main/java/com/facebook/presto/spi/block/LazySliceArrayBlock.java

```diff
@@ -182,8 +183,13 @@ public void assureLoaded()
             throw new IllegalArgumentException("Lazy block loader did not load this block");
         }
 
-        if (dictionary && ids.length != positionCount) {
-            throw new IllegalArgumentException(format("Expected %s positions, loaded %s", positionCount, ids.length));
+        if (dictionary) {
+            if (ids.length != positionCount) {
+                throw new IllegalArgumentException(format("Expected %s positions, loaded %s", positionCount, ids.length));
```

Comment:
It is always valid to call assureLoaded `loader.load()` is when we assign values to the fields of this block. So even though it is not an argument to this method, it is actually the first time values are assigned to the fields. And this is the first place in this class that we can check the "